data-prepper-api/src/test/java/org/opensearch/dataprepper/model/sink/AbstractSinkTest.java

```diff
@@ -69,7 +96,55 @@ public void doOutput(Collection<Record<String>> records) {
 
         @Override
         public void shutdown() {
+            super.shutdown();
+        }
 
+        @Override
+        public void doInitialize() {
+        }
+
+        @Override
+        public boolean isReady() {
+            return true;
+        }
+    }
+
+    private static class AbstractSinkNotReadyImpl extends AbstractSink<Record<String>> {
+
+        boolean initialized;
+        int initCount;
+        public AbstractSinkNotReadyImpl(PluginSetting pluginSetting) {
+            super(pluginSetting);
+            initialized = false;
+            initCount = 0;
+        }
+
+        @Override
+        public void doOutput(Collection<Record<String>> records) {
+            try {
+                Thread.sleep(500);
+            } catch (InterruptedException e) {
+
+            }
+        }
+
+        @Override
+        public void shutdown() {
+            super.shutdown();
+        }
+
+        @Override
+        public void doInitialize() throws Exception {
+            if (++initCount == 3) {
```

Comment:
I think you are trying to make this initialize on the last retry right? You can make the static constant package protected in `AbstractSink` and then use it here. Just remove the `private` modifier there.

If somebody changes that value, this test might end up failing.

Reply:
In `AbstractSink`? That's not test code. This is specific to Testing, right? I can make it static code but this number needs to be smaller than the number of retries I have above, which is 10. I will make these number as static constants with init count less than max retries count.